src/api/weather.js

```diff
@@ -0,0 +1,27 @@
+import axios from "axios";
+
+export const GetWeather = (location) => {
+    const { latitude, longitude } = location;
+
+    const currentParams = "temperature_2m,relative_humidity_2m,precipitation,snowfall,wind_speed_10m";
+
+    return new Promise(async (resolve, reject) => {
```

Comment:
In general you don't need to create a new promise.

Calling `axios.get(...)` is already returning a `Promise`, so in theory you could just have done `return axios.get(...)`.

The only reasons that speaks for doing it as you did here are:
1. Practice :-)
2. The fact that you're re-mapping properties namings in lines 15-19, and it actually makes sense to do it here. If it wasn't for these re-mappings, you could just `return axios.get(...)`.

In general this can stay as is.

I saw you've used `async`/`await`. Do you know what these are, and how they work together with promises - or shall we cover that too?

Reply:
I used it because it was in the documentation. I think both of these simplify working with the promises. The "async" keyword returns an asynchronous function (promise), and "await" controls the promise - it waits until it is either resolved or rejected